LeaderboardBackend/Services/Impl/UserService.cs

```diff
@@ -98,4 +99,23 @@ public async Task<CreateUserResult> CreateUser(RegisterRequest request)
 
         return newUser;
     }
+
+    public async Task<UpdateUserResult> UpdateUser(Guid id, UpdateUserRequest request)
+    {
+        User? user = await applicationContext.Users.SingleOrDefaultAsync(user => user.Id == id);
+
+        if (user is null)
+        {
+            return new UserNotFound();
+        }
+
+        if (user.Role == UserRole.Administrator)
+        {
+            return new BadRole();
+        }
+
+        user.Role = request.Role;
+        await applicationContext.SaveChangesAsync();
+        return new Success();
+    }
```

Comment:
Handle the case where `request.Role` is ADMINISTRATOR or REGISTERED. If you don't then admins are able to unconfirm a user's email and create other admins.